File: 590-n-ary-tree-postorder-traversal/590-n-ary-tree-postorder-traversal.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    List<Integer> result = new ArrayList<>();
    public List<Integer> postorder(Node root) {
       if (root == null) {
           return result;
       } 
       traverse(root);
       result.add(root.val); 
       return result;
    }
    
    private void traverse (Node root)
    {
        if (root == null) {
            return;
        }
        for (Node n : root.children) {
            traverse(n);
            result.add(n.val);
        }
    }
}

File: 590-n-ary-tree-postorder-traversal/590-n-ary-tree-postorder-traversal.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */

/**
 * @param {Node|null} root
 * @return {number[]}
 */
var postorder = function(root) {
    let result=[]
    return postOrdTraversal(root,result)
};

function postOrdTraversal(root,result){
    if(!root)
        return [];
    if(root.children.length){
        root.children.forEach(child=>{
         postOrdTraversal(child,result);
        })
    }
    if(root)
      result.push(root.val)
    return result;
}